python/cudf/cudf/core/column/string.py

```diff
@@ -2035,6 +2039,22 @@ def as_numerical_column(self, dtype, **kwargs):
                     kwargs.update(format=fmt)
         kwargs.update(dtype=out_dtype)
 
+        if str_dtype.kind in ("i"):
+            if not cpp_all_integers(self):
+                bool_check = cpp_is_integer(self)
+                index = bool_check.astype("int8").find_first_value(0)
```

Comment:
I think we have `.any()` and `.all()` so you can check for all True vs all False.

Reply:
Had a discussion offline with @kkraus14 and decided to not have the first error value(non-float/non-int) and instead display a generic error message that we are not able to type-cast due to invalid int/float.